### ISSUE TITLE
Support Not Recognized Custom Class-Validator Decorators

### ISSUE THREAD

**Issue by MR-Mostafa** [darraghoriordan/eslint-plugin-nestjs-typed#279]
Hello,
I'm using `class-validator` in my NestJS application alongside `eslint-plugin-nestjs-typed`. I have created a custom decorator (`IsValidLoginIdentifier`) that internally uses registerDecorator from `class-validator.` My DTO looks like this:

```ts
export class LoginDto {
  @IsValidLoginIdentifier()
  identifier: string;

  @ValidateIf(({ identifier }: LoginDto) => /* ... */)
  @IsString()
  @IsNotEmpty()
  password: string;
}
```

The custom decorator `IsValidLoginIdentifier` is defined like this (simplified):

```ts
@ValidatorConstraint({ /* ... */ })
class IsValidLoginIdentifierConstraint implements ValidatorConstraintInterface {
  validate(value: string) {
    // ...
  }
  // ...
}

export function IsValidLoginIdentifier(validationOptions?: ValidationOptions) {
  return (object: object, propertyName: string) => {
    registerDecorator({
      target: object.constructor,
      propertyName,
      validator: IsValidLoginIdentifierConstraint,
      // ...
    });
  };
}
```

Despite this, the linter reports: (for `identifier` property)

```ts
Property has no class-validator decorator (use @Allow() if you don't need a validation)
eslint@darraghor/nestjs-typed/all-properties-are-whitelisted

Non-optional properties must have a decorator that checks the value is defined (for example: @IsDefined())
eslint@darraghor/nestjs-typed/all-properties-have-explicit-defined
```

It seems like the linter doesn't recognize custom decorators as valid class-validator decorators. If I add or replace my custom decorator with a well-known one (e.g., `@IsString()`), the errors go away. like this:

```ts
export class LoginDto {
  @IsString()
  @IsValidLoginIdentifier()
  identifier: string;

  @ValidateIf(({ identifier }: LoginDto) => /* ... */)
  @IsString()
  @IsNotEmpty()
  password: string;
}
```

Question/Request:

-  Is there a recommended way to configure `eslint-plugin-nestjs-typed` so it recognizes or respects custom `class-validator` decorators?
-  If not, could there be a feature added to allow specifying custom decorator names or patterns that should be treated as valid class-validator decorators?

Thank you for your time and for maintaining this plugin.

**Comment by darraghoriordan** [comment]
Yea those rules don't take settings. You can disable the rule for those decorators or disable it for the whole project.

If you want the feature...

You could submit a pr for custom decorators to be passed as an option. eslint rules allow options to be passed. There is an example of a rule that accepts options params here: https://github.com/darraghoriordan/eslint-plugin-nestjs-typed/blob/main/src/rules/injectablesShouldBeProvided/injectableShouldBeProvided.ts

just add an option that accepts a string[]. set the default to []. append the options provided to the existing list in the rule body. it should be a pretty straightforward PR. It would need tests.
